akka-docs/src/main/paradox/typed/actors.md

```diff
@@ -412,6 +412,24 @@ former simply speaks more languages than the latter. The opposite would be
 problematic, so passing an @scala[`ActorRef[PublishSessionMessage]`]@java[`ActorRef<PublishSessionMessage>`] where
 @scala[`ActorRef[RoomCommand]`]@java[`ActorRef<RoomCommand>`] is required will lead to a type error.
 
+@@@ div {.group-java}
+#### AbstractOnMessageBehavior API
+
+The `AbstractBehavior` API typically makes heavy use of builders which allows `instanceof` checks and casts to be
+performed "behind the scenes".  In some cases, it may be preferred to use a more "direct" style where a `Behavior`
+is, essentially, a function converting a message into a successor `Behavior`.  In contrast to builders, this style
+may have sufficiently reduced runtime overhead to be worth a little extra verbosity (recently-added Java features
+under the "pattern-matching" umbrella may substantially reduce that verbosity).
```

Comment:
Can we phrase this differently so that the new pattern-matching features are highlighted as the primary purpose of this API instead of as a side note?